streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -184,12 +184,12 @@ public void resetProducer() {
         transactionInitialized = false;
     }
 
-    private void maybeBeginTransaction() throws ProducerFencedException {
+    private void maybeBeginTransaction() {
         if (eosEnabled() && !transactionInFlight) {
             try {
                 producer.beginTransaction();
                 transactionInFlight = true;
-            } catch (final ProducerFencedException error) {
+            } catch (final ProducerFencedException | InvalidProducerEpochException error) {
```

Comment:
```suggestion
            } catch (final ProducerFencedException || InvalidProducerEpochException error) {
```

Reply:
This won't work I suppose?

Reply:
Oh lol sorry I was just comparing it with the other block. Obviously nvm